ksqldb-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadataImpl.java

```diff
@@ -105,7 +107,24 @@ public PersistentQueryMetadataImpl(
         maxQueryErrorsQueueSize,
         retryBackoffInitialMs,
         retryBackoffMaxMs,
-        listener
+        new QueryMetadata.Listener() {
```

Comment:
Let just define a QueryMetadata.Listener below as a named class to make this a little cleaner.